sound/soc/sof/pm.c

```diff
@@ -205,8 +212,16 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	target_state = snd_sof_dsp_power_target(sdev);
 
 	/* Skip to platform-specific suspend if DSP is entering D0 */
-	if (target_state == SOF_DSP_PM_D0)
+	if (target_state == SOF_DSP_PM_D0) {
+		ret = snd_sof_dma_trace_trigger(sdev, SNDRV_PCM_TRIGGER_STOP);
```

Comment:
Adding code to this branch makes the comment on L214 harder to understand. Before this patch, this was a simple check to skip to platforms-specific suspend. So maybe modify the comment L214 as well to reflect this.

Reply:
Yes, I will update the comment.

Reply:
why not do 	snd_sof_release_trace(sdev); unconditionally for all target states during suspend?